thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/datalayer/DAOProviderUtilTest.java

```diff
@@ -1,30 +1,29 @@
 package com.linkedin.thirdeye.datalayer;
 
-import com.linkedin.thirdeye.datalayer.bao.AbstractManagerTestBase;
+import com.linkedin.thirdeye.datalayer.bao.DAOTestBase;
 import com.linkedin.thirdeye.datalayer.bao.MergedAnomalyResultManager;
-import com.linkedin.thirdeye.datalayer.bao.jdbc.MergedAnomalyResultManagerImpl;
-import com.linkedin.thirdeye.datalayer.util.DaoProviderUtil;
 import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 
-public class DAOProviderUtilTest extends AbstractManagerTestBase {
+public class DAOProviderUtilTest {
+  private DaoProvider testDAOProvider;
   @BeforeClass
   public void beforeClass() {
-    super.init();
+    testDAOProvider = DAOTestBase.getInstance();
   }
 
   @AfterClass(alwaysRun = true)
   public void afterClass() {
-    super.cleanup();
+    testDAOProvider.restart();
```

Comment:
Sorry to spoil the party, but this will cause concurrency issues. We've been there, done that. TestNG executes @AfterClass and @BeforeClass methods (of different test classes) in parallel.

Reply:
Hi @apucher, Thanks for the comment. Is there a proper solution for that?